Demos/DatAgora_PartDieu/ud-viz-context/BaseDemo.js

```diff
@@ -0,0 +1,1247 @@
+        update: itowns.FeatureProcessing.update,
+        convert: itowns.Feature2Mesh.convert(),
+        source: wfsMaskCSource,
+        style: new itowns.Style({
+          fill:{
+            color: colorEVAArtif,
+            base_altitude : 170,
+          }
+      })
+    });
+
+    this.view.addLayer(wfsMaskCLayer);
+  }
+
+  /**
+   * Create an iTowns 3D Tiles layer based on the specified layerConfig.
+   * @param {string} layerConfig The name of the layer to setup from the
+   * generalDemoConfig.json config file (should be one of the properties
+   * of the 3DTilesLayer object in
+   * UD-Viz/UD-Viz-Core/examples/data/config/generalDemoConfig.json
+   * config file).
+   */
+  
+  /**
+   * The configuration file of demo defines URL of resources. At the time
+   * of the writing of the configuration of the demo the Fully Qualified Domain 
+   * Name (FQDN) of the server hosting the demo is not yet known (and will
+   * only be known at deployment time). It is thus configured with the 
+   * 'DEMO_SERVER' keyword (in the configuration file) that is replaced (at 
+   * run time) through the usage of this function.
+   * This version (as opposed to ReconfigureUrlWithPort) does NOT include
+   * the port number.
+   * @param  {string} ConfigUrl The url of the resource as read in the 
+   *                            configuration file
+   * @return {string}           The reconfigured URL.
+   * This is a private method (hard to express in JS!?).
+   */
+   static ReconfigureUrlWithoutPort(ConfigUrl) {
+    // The url of the 3DTiles layer as read in the configuration file
+    if (!ConfigUrl.includes('DEMO_SERVER')) {
+      // Nothing to fix, return the original url
+      return ConfigUrl;
+    }
+
+    // The url of the server that served this javascript source
+    const ServerUrl = `${window.location.protocol}//${window.location.hostname}`;
+    // The actual URL of the layer as hosted on the (same) server that
+    // delivered this javascript source
+    return ConfigUrl.replace('DEMO_SERVER', ServerUrl);
+  }
+
+  /**
+   * Refer to the documentation of ReconfigureUrlWithoutPort(ConfigUrl)
+   * for the role, parameter and return value.
+   * This version (as opposed to ReconfigureUrlWithoutPort) DOES include
+   * the port number.
+   */
+  static ReconfigureUrlWithPort(ConfigUrl) {
+    // The url of the 3DTiles layer as read in the configuration file
+    if (!ConfigUrl.includes('DEMO_SERVER')) {
+      // Nothing to fix, return the original url
+      return ConfigUrl;
+    }
+
+    // The url of the server that served this javascript source
+    let ServerUrl = `${window.location.protocol}//${window.location.hostname}`;
+    ServerUrl += (window.location.port ? ':' + window.location.port : '');
+    // The actual URL of the layer as hosted on the (same) server that
+    // delivered this javascript source
+    return ConfigUrl.replace('DEMO_SERVER', ServerUrl);
```

Comment:
Are these lines still relevant now that the config file is configured using the .env file? Should we remove them in demos?

Reply:
When code is not used, clean it up. Or it will confuse the next reader...